src/autopas/containers/directSum/DirectSum.h

```diff
@@ -267,13 +305,26 @@ class DirectSum : public CellBasedParticleContainer<FullParticleCell<Particle>>
   }
 
   bool deleteParticle(Particle &particle) override {
+    // swap-delete helper function
+    auto swapDelFromVector = [&](auto &particleVec) -> bool {
+      const bool isRearParticle = &particle == &particleVec.back();
+      particle = particleVec.back();
+      particleVec.pop_back();
+      return isRearParticle;
+    };
+
     // deduce into which vector the reference points
-    auto &particleVec = particle.isOwned() ? getCell()._particles : getHaloCell()._particles;
-    const bool isRearParticle = &particle == &particleVec.back();
-    // swap-delete
-    particle = particleVec.back();
-    particleVec.pop_back();
-    return not isRearParticle;
+    if (particle.isOwned()) {
+      return swapDelFromVector(getOwnedCell()._particles);
+    } else if (particle.isHalo()) {
+      for (auto cellIt = ++this->_cells.begin(); cellIt != this->_cells.end(); cellIt++) {
```

Comment:
I implemented this position-based search.

While looking at it, I noticed that this method (`bool deleteParticle(Particle &particle)`) is never really used and seems a bit unsafe. E.g. if you call it with a particle that is not part of the container.
Do you know what the idea of this method was? Else I will add an issue.

Reply:
Regarding `bool deleteParticle(Particle &particle)`:
The idea is that you can delete particles without having to create and point an iterator at them. We actually use this in LADDS in a few places because we get the references in our pairwise interaction, which is the collision detection.